modules/redis/options.go

```diff
@@ -0,0 +1,141 @@
+package redis
+
+import (
+	"crypto/tls"
+	"net"
+	"strconv"
+
+	"github.com/mdelapenya/tlscert"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+type options struct {
+	tlsPort          string
+	withSecureURL    bool
+	withMTLSDisabled bool
+	tlsConfig        *tls.Config
+}
+
+// Compiler check to ensure that Option implements the testcontainers.ContainerCustomizer interface.
+var _ testcontainers.ContainerCustomizer = (Option)(nil)
+
+// Option is an option for the Redpanda container.
+type Option func(*options) error
+
+// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
+func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// WithTLS sets the TLS configuration for the redis container, setting
+// the port to listen on for TLS connections, whether to use a secure URL,
+// and whether to disable MTLS (mutual TLS).
+func WithTLS(tlsPort string, secureURL bool, mtlsDisabled bool) Option {
+	return func(o *options) error {
+		o.tlsPort = tlsPort
+		o.withSecureURL = secureURL
+		o.withMTLSDisabled = mtlsDisabled
+		return nil
+	}
+}
+
+// WithConfigFile sets the config file to be used for the redis container, and sets the command to run the redis server
+// using the passed config file
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
```

Comment:
suggestion: pass in an io.Reader which makes it easier to use with the embed module.

Reply:
Same here regarding BCs: this function already existed prior to this PR